### PR TITLE
fix(transport): Use correct data category for transaction events

### PR DESCRIPTION
**Pull request by untitaker**
While working on https://github.com/getsentry/sentry-python/pull/825 I found a bug that caused us to use the "default" item type for transaction events. The bug happened because there were mismatched expectations as to what Item.get_event should return if the content is a transaction event.

Will add an explicit test later.

### CODE REVIEW

File: sentry_sdk/envelope.py [hunk 1]
```diff
@@ -230,23 +230,25 @@ def __repr__(self):
...
 
     def get_bytes(self):
         # type: (...) -> bytes
         return self.payload.get_bytes()
 
     def get_event(self):
         # type: (...) -> Optional[Event]
-        if self.headers.get("type") == "event" and self.payload.json is not None:
+        """
+        Returns a regular event if there is one. Transaction events are exempt
```
Comment (rhcarvalho) [inline sentry_sdk/envelope.py:248]: What do you mean by "transaction events are exempt from this"?

File: tests/test_transport.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ def test_simple_rate_limits(httpserver, capsys, caplog, make_client):
     client.capture_event({"type": "transaction"})
     client.flush()
 
-    assert len(httpserver.requests) == 1
+    assert list(r.url[-18:] for r in httpserver.requests) == ["/api/132/envelope/"]
```
Comment (rhcarvalho) [inline tests/test_transport.py:119]: `url[-18:]` :-(

I'd keep the original check for 1 request, then you can `assert httpserver.requests[0].endswith("/api/132/envelope/"]`

**Review by rhcarvalho — CHANGES_REQUESTED** [review]
At least one bug found :)

File: tests/test_transport.py (hunk 2 above)
Comment (untitaker) [reply tests/test_transport.py:119]: yeah fair

File: sentry_sdk/envelope.py (hunk 1 above)
Comment (untitaker) [reply sentry_sdk/envelope.py]: Nevermind. I reverted the changes to this method as they were not necessary.

File: sentry_sdk/envelope.py [hunk 3]
```diff
@@ -230,22 +223,27 @@ def __repr__(self):
...
+            return "default"
 
     def get_bytes(self):
         # type: (...) -> bytes
         return self.payload.get_bytes()
 
     def get_event(self):
         # type: (...) -> Optional[Event]
+        """
+        Returns a regular (non-transaction) event if there is one.
```
Comment (rhcarvalho) [inline sentry_sdk/envelope.py:245]: "regular non-transaction event" is weird. Does this mean

```suggestion
        Returns an error event if there is one.
```

?

**Review by rhcarvalho — APPROVED** [review]

**Pull request merged**